apps/src/tutorialExplorer/tutorialDetail.jsx

```diff
@@ -42,6 +50,44 @@ export default class TutorialDetail extends React.Component {
     }
   };
 
+  focusIn = event => {
+    // for starters, if our scroll has changed since the dialog opened, then reset it to where we were at
+    // the beginning. Otherwise, if you shift-tab above the detail on the first tutorial or tab below the detail
+    // on a tutorial towards the end of the window, it'll fidget the scroll. This keeps it consistent.
+    if (window.scrollY !== this.startingScroll[1]) {
+      window.scrollTo(...this.startingScroll);
+    }
+
+    /*
+      We've kept a ref to our dialog container, so when the user is adjusting focus we look to see if we're
+      focusing onto an element that's still in the modal. If so, then awesome! We're done. If not, then we look
+      in our modal and find the first button (which is the X to close) or the last link (which could vary based upon
+      what's in there).
+
+      If we're shifting focus away from the first button (and have left the modal), then we wrap around to the last link.
+      Otherwise, if we're leaving, we return to the start at the first button.
+
+      This feels very clunky. :-(
```

Comment:
Not a fan of shipping something we are disappointed about.  This UI is very high usage and we should be highly confident in our approach.

Reply:
Just because I said it was clunky didn't mean I'm not confident in the approach. 😄 

This is the accepted way to do it when rolling your own approach, it's just always annoying when you have to hop outside of react and directly deal with dom manipulations instead of handling it all in the component hierarchy itself. 

Doing it all inside of react would require substantial additional refactoring to all of the components in here, which I felt was better to defer atm.